src/osd/PrimaryLogPG.cc

```diff
@@ -7944,6 +8004,10 @@ inline int PrimaryLogPG::_delete_oid(
     dout(20) << __func__ << " setting whiteout on " << soid << dendl;
     oi.set_flag(object_info_t::FLAG_WHITEOUT);
     ctx->delta_stats.num_whiteouts++;
```

Comment:
This if and the case below at line 8030 are identical.  You should probably pull them out to a single one right before the if(whiteout) block.  Semantically, this is because this little bit of behavior is identical whether the object lives on as a whiteout or not (in either case, we need to decrement the refs and the num_oibjects_manifest count).